pyomo/core/tests/unit/test_kernel_register_numpy_types.py

```diff
@@ -16,7 +16,10 @@
 # Boolean
 numpy_bool_names = []
 if numpy_available:
-    numpy_bool_names.append('bool_')
+    if numpy.__version__[0] == '2':
+        numpy_bool_names.append('bool')
+    else:
+        numpy_bool_names.append('bool_')
```

Comment:
Why do you check version here, but everywhere else you use `hasattr`?

Reply:
Because John added that one...